app/services/change_schedule.rb

```diff
@@ -131,6 +131,71 @@ def update_ecf_records!
 
     update_school_cohort_and_schedule!
     update_induction_records!
+    update_historical_induction_records!
+  end
+
+  def in_target?(induction_record)
+    in_target_cohort?(induction_record) && in_target_schedule?(induction_record)
+  end
+
+  def in_target_cohort?(induction_record)
+    induction_record.cohort_start_year == new_schedule.cohort_start_year
+  end
+
+  def in_target_schedule?(induction_record)
+    induction_record.schedule == new_schedule
+  end
+
+  def historical_target_school_cohort(historical_record)
+    historical_record.school.school_cohorts.for_year(new_schedule.cohort_start_year).first.presence || SchoolCohort.create!(school: historical_record.school,
+                                                                                                                            cohort: new_schedule.cohort,
+                                                                                                                            induction_programme_choice: historical_record.induction_programme&.training_programme)
+  end
+
+  def historical_school_cohort_partnership(historical_record)
+    Partnership.create_with(relationship: true)
+    .find_or_create_by!(school: historical_record.school,
+                        cohort: new_schedule.cohort,
+                        lead_provider: historical_record.lead_provider,
+                        delivery_partner: historical_record.delivery_partner)
+  end
+
+  def historical_induction_programme(historical_record)
+    return historical_record.induction_programme if in_target_cohort?(historical_record)
+
+    historical_school_cohort = historical_target_school_cohort(historical_record)
+
+    set_default_programme = !historical_record.enrolled_in_fip? || !historical_record.lead_provider_id || !historical_record.delivery_partner_id
+
+    if set_default_programme
+      historical_school_cohort.default_induction_programme.presence || InductionProgramme.create_with(partnership: historical_record.partnership)
```

Comment:
I think we get to this point if the participant does not belong to a lead provider/ is in CIP, so we shouldn't be creating a partnership unless I misunderstood something?